service/forward_service.cc

```diff
@@ -462,6 +476,10 @@ future<query::forward_result> forward_service::execute_on_this_shard(
         // Execute query.
         while (!pager->is_exhausted()) {
             co_await pager->fetch_page(rs_builder, DEFAULT_INTERNAL_PAGING_SIZE, now, timeout);
+
+            if (_shutdown) {
+                throw std::runtime_error("forward_service is shutting down");
```

Comment:
And here

Reply:
Why? I thought that your previous comment pointed out that throwing in a non-coroutine lambda may have surprising effects. This method is a coroutine, could you elaborate on what is wrong here?